test/test_other.py

```diff
@@ -7691,6 +7691,136 @@ def test_ld_library_path(self, args):
     self.assertContained('Hello4', out)
     self.assertContained('Ok', out)
 
+  @also_with_wasmfs
+  def test_ld_library_path_dependencies(self):
```

Comment:
Can we call this `test_dlopen_rpath` maybe?   (at least it should start with something like `test_dlopen` or `test_dylink`).

Also, perhaps we have an existing tests which has shared libraries with dependencies that we can make this a variant of?